### PR TITLE
Fix some coverity warnings.

### PR DESCRIPTION
**Pull request by godfath3r**
Fix some coverity scan issues.

### CODE REVIEW

File: core/src/main/java/hudson/model/RunParameterDefinition.java [hunk 1]
```diff
@@ -148,6 +148,9 @@ public String getHelpFile() {
 
         @Override
         public ParameterDefinition newInstance(StaplerRequest req, JSONObject formData) throws FormException {
+            if (req == null)
```
Comment (KostyaSha) [inline core/src/main/java/hudson/model/RunParameterDefinition.java]: Or you can require Nonnull for argument.

File: core/src/main/java/hudson/util/io/RewindableRotatingFileOutputStream.java [hunk 2]
```diff
@@ -55,7 +59,9 @@ public void rewind() throws IOException {
             File fi = getNumberedFileName(i);
             if (fi.exists()) {
                 File next = getNumberedFileName(i+1);
-                next.delete();
+                if (!next.delete()){
+                    throw new IOException(String.format("Could not delete %s", next.getAbsolutePath()));
```
Comment (KostyaSha) [inline core/src/main/java/hudson/util/io/RewindableRotatingFileOutputStream.java]: Why do you think that this behaviour is right?

**Comment by KostyaSha** [comment]
bcc @oleg-nenashev

File: core/src/main/java/hudson/util/io/RewindableRotatingFileOutputStream.java [hunk 3]
```diff
@@ -66,7 +70,9 @@ public void rewind() throws IOException {
      */
     public void deleteAll() {
         for (int i=0; i<=size; i++) {
-            getNumberedFileName(i).delete();
+            File f= getNumberedFileName(i);
+            if (!f.delete())
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/util/io/RewindableRotatingFileOutputStream.java]: Better to replace it by the new `Files` API

File: core/src/main/java/jenkins/install/InstallUtil.java [hunk 4]
```diff
@@ -318,7 +318,9 @@ private static String getCurrentExecVersion() {
     public static synchronized void persistInstallStatus(List<UpdateCenterJob> installingPlugins) {
         File installingPluginsFile = getInstallingPluginsFile();
 	if(installingPlugins == null || installingPlugins.isEmpty()) {
-		installingPluginsFile.delete();
+		if (!installingPluginsFile.delete()) {
+		    LOGGER.log(SEVERE, String.format("Failed to delete %s", installingPluginsFile.getAbsolutePath()));
```
Comment (oleg-nenashev) [inline core/src/main/java/jenkins/install/InstallUtil.java]: `Files` API as well. In such case the error cause will be propagated as exception

File: core/src/main/java/jenkins/install/UpgradeWizard.java [hunk 5]
```diff
@@ -143,8 +144,10 @@ public HttpResponse doSnooze() throws IOException {
         Jenkins.getInstance().checkPermission(Jenkins.ADMINISTER);
         File f = SetupWizard.getUpdateStateFile();
         FileUtils.touch(f);
-        f.setLastModified(System.currentTimeMillis() + TimeUnit.DAYS.toMillis(1));
-        LOGGER.log(FINE, "Snoozed the upgrade wizard notice");
+        if (f.setLastModified(System.currentTimeMillis() + TimeUnit.DAYS.toMillis(1)))
```
Comment (oleg-nenashev) [inline core/src/main/java/jenkins/install/UpgradeWizard.java]: `Files` API again. Brackets would be useful as well

File: core/src/main/java/hudson/model/RunParameterDefinition.java [hunk 6]
```diff
@@ -148,6 +148,9 @@ public String getHelpFile() {
...
+                throw new FormException("ParameterDefinition new instance method is called for null Stapler request.",
+                        "req");
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/RunParameterDefinition.java]: IMHO no need in the line break. Brackets would be preferable

**Comment by godfath3r** [comment]
I've added some checks for null state in DefaultJnlpSlaveReceiver.java
We can squash the commits on the merge.

I will also work on your comments.

**Comment by oleg-nenashev** [comment]
@godfath3r any plans to finish it?

**Comment by godfath3r** [comment]
@oleg-nenashev not really. I think coverity fixes should be done one by one, in case some some discussion/review should be done. I'm closing this for now.

**Pull request closed without merging**